doc/source/development/extending.rst

```diff
@@ -208,13 +208,201 @@ will
+operations: :meth:`Series.any`, :meth:`Series.all`,
+:meth:`Series.min`, :meth:`Series.max`, :meth:`Series.sum`,
+:meth:`Series.mean`, :meth:`Series.median`, :meth:`Series.prod`
+(and its alias :meth:`Series.product`), :meth:`Series.std`,
+:meth:`Series.var`, :meth:`Series.sem`,
+:meth:`Series.kurt`, and :meth:`Series.skew`.
+
+In order to implement any of these functions, your ExtensionArray must include
+an implementation of :meth:`ExtensionArray._reduce`. Once you provide an
+implementation of :meth:`ExtensionArray._reduce` which handles a particular
+method, calling the method on the Series will invoke the implementation
+on your ExtensionArray. All these methods are reduction functions, and
+so are expected to return a scalr value of some type. However it is perfectly
+acceptable to return some instance of an :class:`pandas.api.extensions.ExtensionArray`.
+
+Series operations which are not handled by :meth:`ExtensionArray._reduce`,
+such as :meth:`Series.round`, will generally invoke an equivalent numpy
+function with your extension array as the argument. Pandas only guarantees
+that your array will be passed to a numpy function, it does not dictate
+how your ExtensionArray should interact with numpy's dispatch logic
+in order to achieve its goal, since there are several alternative ways
+of achieving similar results.
+
+However, the details of numpy's dispatch logic are not entirely simple, and
+there nuances which you should be aware of. For that reason, and in order to
+make it easier to create new pandas extensions, we will now cover some of
+possible approaches of dealing with numpy.
+
+The first alternative, and the simplest, is to simply provide an `__array__`
+method for your ExtensionArray. This is a standard numpy function documented
+here (TBD), which must return a numpy array equivalent of your ExtensionArray.
+This will usually be an array whose dtype is `object` and whose values are
+instances of some class which your ExtensionArray wraps into an array. For
+example, the pandas tests include an ExtensionArray example called
+`DecimalArray`, if it used this method, its `__array__` method would return an
+ndarray of `decimal.Decimal` objects.
+
+Implementing `__array__`  is easy, but it usually isn't satisfactory because
+it means most Series operations will return a Series of object dtype, instead
+of maintaining your ExtensionArray's dtype.
+
+The second approach is more involved, but it does a proper job of maintaining
+the ExtensionArray's dtype through operations. It requires a detailed
+understanding of how numpy functions operate on non ndarray objects.
+
+Just as pandas handles some operation via :meth:`ExtensionArray._reduce`
+and others by delegating to numpy, numpy makes a distinction between
+between two types of operations: ufuncs (such as `np.floor`, `np.ceil`,
+and `np.abs`), and non-ufuncs (for example `np.round`, and `np.repeat`).
+
+We will deal with ufuncs first. You can find a list of numpy's ufuncs here
```

Comment:
Can we simplify this? How about

1. State that Ufuncs called on a Series will unwrap the array, call the ufunc on the array, and wrap the results.
2. Implementing `__array_ufunc__` gives you control over how the ufunc operates on your data.
3. For example, look at SparseArray. It calls the ufunc twice: once on the sparse values and once on the fill value, and constructs a new SparseArray without having materialized the full, dense array in memory (link to implementation).

Reply:
Has (1) been merged yet? I thought that was #23293 and still pending, 8 months old now. 

It's important to link to a list, since np doesn't make it clear what functions ufuncs and which aren't (I didn't know offhand for one).

 I deleted the second half of the paragraph. Maybe that's enough.